x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateParse.java

```diff
@@ -40,7 +44,13 @@ public class DateParse extends ScalarFunction implements OptionalArgument, Evalu
     private final Expression field;
     private final Expression format;
 
-    public DateParse(Source source, Expression first, Expression second) {
+    @Described("Parses a string into a date value")
+    @Typed("date")
+    public DateParse(
+        Source source,
+        @Named("datePattern") @Typed("keyword") @Described("A valid date pattern") @Optional Expression first,
+        @Named("dateString") @Typed("keyword") @Described("A string representing a date") Expression second
+    ) {
```

Comment:
Instead of having multiple annotations (which can have different order, might be missing, etc..) use one with multiple fields:
`@Param(name="datePattern", type="keyword", description="A valid date pattern", optional=true/false)`

Separately it's worth checking if the field name is not already available in the debug info (method.getParameterNames()) - this would avoid the `name` attribute and use the field name instead.
This will force us to actually give them better names such as datePattern instead of first.

Reply:
👍 

Debug info could not be available in all the cases, but I'll give it a try